### DISCUSSION TITLE
Next.js (14.2.3)  Unhandled Runtime Error Error: Hydration failed because the initial UI does not match what was rendered on the server.

### DISCUSSION THREAD

**Discussion by Pipps01** [vercel/next.js#66465] [Help]
### Summary

![1](https://github.com/vercel/next.js/assets/167841166/7319681d-d7f1-40aa-a191-8b4c97b7292c)
![2](https://github.com/vercel/next.js/assets/167841166/0804d850-90e1-4355-be84-2be5b7666559)
![3](https://github.com/vercel/next.js/assets/167841166/a01ee154-4335-4523-be11-0c97673ba016)
![4](https://github.com/vercel/next.js/assets/167841166/70f260aa-5c40-4ad9-b4da-d5e62f2a6e49)
![5](https://github.com/vercel/next.js/assets/167841166/3b15c7f1-41aa-4d10-982e-6cc7742531bf)

### Additional information

```js
Hello, I hope you can help me 🥺
```

**Comment by serdar-webdev** [comment]
Hi folk, you can fix it with not better solution but it will work, you can disable ssr for the component where this error occured, because of it's not matter are you use 'use client' it will try to render it on server, the result of server and client matches different content.
const ClientComponent = dynamic(() => import('../components/CodeSampleModal'), {
  ssr: false,
});

like this. You can use it before you find the error

**Reply by Pipps01** [reply]
I don't know what I'm doing wrong, but the error persists.

**Comment by serdar-webdev** [comment]
And it's not my business but, you should use server components to fetch data, it's the point of next js, if you decrease codes of client component it can solve the problem
